src/transformers/convert_slow_tokenizer.py

```diff
@@ -1552,6 +1552,35 @@ def converted(self) -> Tokenizer:
         return tokenizer
 
 
+class LlamaTikTokenConverter(TikTokenConverter):
+    def tokenizer(self):
+        vocab_scores, merges = self.extract_vocab_merges_from_model(self.vocab_file)
+        tokenizer = Tokenizer(BPE(vocab_scores, merges, fuse_unk=False))
+        if hasattr(tokenizer.model, "ignore_merges"):
+            tokenizer.model.ignore_merges = True
+        num_reserved_special_tokens = 256
+        special_tokens = [
+            "<|begin_of_text|>",
+            "<|end_of_text|>",
+            "<|reserved_special_token_0|>",
+            "<|reserved_special_token_1|>",
+            "<|reserved_special_token_2|>",
+            "<|reserved_special_token_3|>",
+            "<|start_header_id|>",
+            "<|end_header_id|>",
+            "<|reserved_special_token_4|>",
+            "<|eot_id|>",
+            "<|python_tag|>",  # end of turn
+        ] + [f"<|reserved_special_token_{i}|>" for i in range(5, num_reserved_special_tokens - 5)]
```

Comment:
let's go a little bit deeper here, `disallowed_special` and `allow_special` exist in tiktoken, can't we extract special tokens like we do for spm ?

Reply:
Agreed that would be nice! Looking into it [here](https://github.com/openai/tiktoken/blob/c0ba74c238d18b4824c25f3c27fc8698055b9a76/tiktoken/core.py#L89), `allowed_special` and `disallowed_special` are params given by the user to control whether `special_tokens` are allowed to be encoded. `allowed_special='all' `would include all of .special_tokens_set. However, it looks like this set is[ hardcoded independently](https://github.com/openai/tiktoken/blob/c0ba74c238d18b4824c25f3c27fc8698055b9a76/tiktoken_ext/openai_public.py#L30) of the file link used for `load_tiktoken_bpe`, so I don't think it can be extracted from the `tiktoken_file` alone